src/main/kotlin/org/rust/lang/core/psi/RustTokenType.kt

```diff
@@ -1,10 +1,36 @@
 package org.rust.lang.core.psi
 
+import com.intellij.lang.ASTNode
+import com.intellij.lang.DefaultASTFactory
+import com.intellij.openapi.components.service
+import com.intellij.psi.impl.source.tree.LeafPsiElement
 import com.intellij.psi.tree.IElementType
+import com.intellij.psi.tree.ILeafElementType
 import org.rust.lang.RustLanguage
+import org.rust.lang.core.psi.impl.IRustLiteralImplBase
 
-open class RustTokenType(debugName: String) : IElementType(debugName, RustLanguage)
+private val defaultASTFactory by lazy { service<DefaultASTFactory>() }
+
+open class RustTokenType(debugName: String) : IElementType(debugName, RustLanguage), ILeafElementType {
+    override fun createLeafNode(leafText: CharSequence?): ASTNode = LeafPsiElement(this, leafText)
```

Comment:
It's better to change signature to ` createLeafNode(leafText: CharSequence)`. There is no nullability annotations on the Java side, so both signutares are valid. And because we assume that it is not `null`( `!!` in children classes), it's better to state our assumption upfront